src/Hadolint/Rules.hs

```diff
@@ -1184,3 +1185,50 @@ noIllegalInstructionInOnbuild = instructionRule code severity message check
     check (OnBuild (From _)) = False
     check (OnBuild (Maintainer _)) = False
     check _ = True
+
+noSelfreferencingEnv :: Rule
+noSelfreferencingEnv = instructionRuleState code severity message check []
```

Comment:
Please use `Set` for the state. That way you avoid calling `nub`

Reply:
Done. The state is a Set now.